src/main/java/teammates/logic/api/AccessControlUtil.java

```diff
@@ -1,5 +1,239 @@
+            String section = "";
+            StudentAttributes student = logic.getStudentForEmail(courseId, recipient);
+            if (student != null) {
+                section = student.section;
+            }
+            return instructor.isAllowedForPrivilege(section, privilegeName);
+        } else {
+            // TODO: implement this if instructor is later allowed to be added
+            // to recipients
+            return false;
+        }
+    }
+
+    /**
+     * Returns a {@code List} of boolean values representing the permissions a student has for a given comment.
+     */
+    public List<Boolean> getVisibilityPermissionsOnCommentForStudent(
+            CommentAttributes comment, StudentAttributes student, List<String> teammatesEmails,
+            List<String> sectionStudentsEmails, List<String> teamsInSection) {
+        
+        List<Boolean> permissions = new ArrayList<Boolean>();
+        for (int i = 0; i < COMMENT_PERMISSIONS_TOTAL_SIZE; i++) {
+            permissions.add(false);
+        };
+        
+        if (comment == null || student == null) {
+            return permissions;
+        }
+        
+        // the following if-else if-else blocks rely on their ordering to properly classify the comment
+        // eg. a comment directed at the student's team will not be classified as "in the same section as student"
+        
+        // individual student level
+        if (isStudentRecipientOfComment(comment, student)) {
+            permissions = getPermissionsForComment(comment, CommentParticipantType.PERSON, true);
+            
+        // team level
+        } else if (isStudentTeamRecipientOfComment(comment, student)) {
+            permissions = getPermissionsForComment(comment, CommentParticipantType.TEAM, true);
+        } else if (isStudentInSameTeamAsRecipientOfComment(comment, student, teammatesEmails)) {
+            permissions = getPermissionsForComment(comment, CommentParticipantType.TEAM, false);
+
+        // section level
+        } else if (isStudentSectionRecipientOfComment(comment, student)) {
+            permissions = getPermissionsForComment(comment, CommentParticipantType.SECTION, true);
+        } else if (isStudentInSameSectionAsRecipientOfComment(comment, student, sectionStudentsEmails, teamsInSection)) {
+            permissions = getPermissionsForComment(comment, CommentParticipantType.SECTION, false);
 
+        // course level
+        } else if (isStudentCourseRecipientOfComment(comment, student)) {
+            permissions = getPermissionsForComment(comment, CommentParticipantType.COURSE, true);
+        } else if (isStudentInSameCourseAsRecipientOfComment(comment, student)) {
+            permissions = getPermissionsForComment(comment, CommentParticipantType.COURSE, false);
+            
+        // comment not in same course as student, so student should not be able to view the comment
+        } else {
```

Comment:
If there is nothing in the `else` block, just remove it. I think this will be caught by either Checkstyle or PMD.